### PR TITLE
Resolves Issue #161

### PR DESCRIPTION
**Pull request by MeikTranel**

### CODE REVIEW

File: src/Grapevine/Server/Route.cs [hunk 1]
```diff
@@ -234,10 +234,11 @@ public override int GetHashCode()
             var parsed = new Dictionary<string, string>();
             var idx = 0;
 
-            foreach (var match in PathInfoPattern.Matches(pathinfo).Cast<Match>().Where(m => m.Success))
+            var matches = PathInfoPattern.Matches(pathinfo)[0].Groups;
+            for (int i = 1; i <= matches.Count-1; i++)
```
Comment (scottoffen) [inline src/Grapevine/Server/Route.cs]: Can change this to `i < matches.Count`.

**Comment by scottoffen** [comment]
Great catch, thanks! If you could make that one change, I'll merge it on it.

Typically, we would ask for some updated test cases, but it appears that while our code coverage tool told us we had 100% coverage, we actually do not have any test cases that explicitly test this method. Our bad. We'll open a separate issue to resolve that, unless you want to tackle that here, too.

**Comment by MeikTranel** [comment]
Honestly i don't know how the hell to properly accept change requests on Github. i had assumed commiting the changes would suffice.


On another note:
If theres mention that multiple parameters can be used in routes, someone should probably add an example on how to properly work with multiple params or conflicting routes for that matter.
Right now as a layman it looks like if you have two routes like this: 

`
Baseroute: "/departments" {
  Route: "/subdepartments",
  Route: "/subdepartments/users"
}
`
or like this: (atlassian jira rest api)
https://docs.atlassian.com/jira/REST/cloud/#api/2/comment/{commentId}/properties

Right now the only obvious way of preventing that /subdepartments route executed when calling the /subdepartments/users route is by having the input path string checked for additional paths in every single routepath that is a substring of another route path.

**Comment by scottoffen** [comment]
The reviewer approves the changes - which I just did, so it's all taken care of now.

As for multiple parameters and conflicting routes, we definitely needs some docs around that. That documentation is in the works. The good news is that the routes as you described them above won't conflict at all.  Consider the following:

```csharp
using System;
using Grapevine.Interfaces.Server;
using Grapevine.Server;
using Grapevine.Server.Attributes;
using Grapevine.Shared;

namespace Grapevine.Local
{
    public class Program
    {
        public static void Main(string[] args)
        {
            using (var server = new RestServer())
            {
                server.LogToConsole();
                server.AfterStarting += restServer => { Console.WriteLine("#-------------------- Server Started --------------------#"); };
                server.AfterStopping += restServer => { Console.WriteLine("#-------------------- Server Stopped --------------------#"); };

                server.Start();
                Console.ReadLine();

                server.Stop();
                Console.ReadLine();
            }
        }
    }

    [RestResource(BasePath = "/departments")]
    public class TestResource
    {
        [RestRoute(HttpMethod = HttpMethod.GET, PathInfo = "/subdepartments")]
        public IHttpContext Subdepartments(IHttpContext context)
        {
            context.Response.SendResponse("Subdepartments");
            return context;
        }

        [RestRoute(HttpMethod = HttpMethod.GET, PathInfo = "/subdepartments/users")]
        public IHttpContext SubdepartmentsUsers(IHttpContext context)
        {
            context.Response.SendResponse("Subdepartments Users");
            return context;
        }
    }
}
```

No matter what the order is of the routes in the class, they won't conflict. Even if you added them manually, they won't conflict. That's because neither one has any parameters or regular expression matches.

If they did conflict, whichever one is registered first will execute first, and if that one responds to the request, by default no other routes will respond.

**Comment by MeikTranel** [comment]
hey @scottoffen now that its merged into master, can you push this onto the nuget release channel? my colleague is using grapevine and desperately waits for this fix to land on his project. ;)

**Comment by scottoffen** [comment]
Sure. We have a release scheduled for Friday, and one more issue that we are still trying to squeeze in, but I can push an interim nuget with the current changes out today.

**Comment by scottoffen** [comment]
Published [4.0.0.235-alpha](https://www.nuget.org/packages/Grapevine/4.0.0.235-alpha)

**Pull request merged**